frontend/src/pages/ProjectSettings/ServicesTable/ServicesTable.tsx

```diff
@@ -0,0 +1,288 @@
+import { Button } from '@components/Button'
+import LoadingBox from '@components/LoadingBox'
+import {
+	useEditServiceGithubSettingsMutation,
+	useGetServicesLazyQuery,
+} from '@graph/hooks'
+import { namedOperations } from '@graph/operations'
+import {
+	Badge,
+	Box,
+	Combobox,
+	IconSolidCubeTransparent,
+	IconSolidExternalLink,
+	IconSolidGithub,
+	Stack,
+	Table,
+	Text,
+	useComboboxState,
+} from '@highlight-run/ui'
+import { useGitHubIntegration } from '@pages/IntegrationsPage/components/GitHubIntegration/utils'
+import { useParams } from '@util/react-router/useParams'
+import { capitalize } from 'lodash'
+import { debounce } from 'lodash'
+import React, { useEffect, useMemo, useState } from 'react'
+
+import { Service, ServiceStatus } from '@/graph/generated/schemas'
+
+import { GitHubSettingsModal } from '../GitHubSettingsModal/GitHubSettingsModal'
+import * as styles from './ServicesTable.css'
+
+type Pagination = {
+	after?: string
+	before?: string
+}
+
+export const ServicesTable: React.FC = () => {
+	const { project_id } = useParams<{ project_id: string }>()
+	const [loadServices, { error, data, loading }] = useGetServicesLazyQuery()
+	const [query, setQuery] = useState<string>('')
+	const [pagination, setPagination] = useState<Pagination>({})
+	const [selectedService, setService] = useState<Service | null>(null)
+
+	const {
+		settings: { isIntegrated },
+		data: githubData,
+	} = useGitHubIntegration()
+
+	const [editServiceGithubSettings] = useEditServiceGithubSettingsMutation()
+
+	useEffect(() => {
+		loadServices({
+			variables: {
+				project_id: project_id!,
+				query: query!,
```

Comment:
I don't think you need the `!` for query, before, and after since they're optional